src/preparser.js

```diff
@@ -5,45 +5,28 @@ const windows = (os.platform() === 'win32');
 
 // Replace out env variables.
 const parseEnvVariables = function (input) {
-  const regex1 = (windows) ?
-    /(\%.*?\%)/ :
-    /(\${[^\$]*}|\$[^\$]*)/;
-  const regex2 = (windows) ?
-    /^\%|\%$/g :
-    /^\${|}$|^\$/g;
+  const referenceRegex =
+    /\${([a-zA-Z_][a-zA-Z0-9_]*)}|\$([a-zA-Z_][a-zA-Z0-9_]*)/g;
 
-  let total = '';
-  function iterate(str) {
-    const match = regex1.exec(str);
-    if (match !== null) {
-      const string = match[0];
-      const stripped = String(string.replace(regex2, '')).toLowerCase();
-      let value = null;
+  return input.replace(referenceRegex, function (varRef, capture1, capture2, capture3) {
+    const varName = capture1 || capture2 || capture3;
+    let value = '';
+    if (windows) {
       for (const name in process.env) {
         if (process.env.hasOwnProperty(name)) {
-          if (String(name).toLowerCase() === stripped) {
+          // Windows is case insensitive
```

Comment:
So I just checked on my Windows box, and it looks like `process.env.path` is identical to `process.env.PATH` (I tried a few other capitalization combos). So it looks like we'll get the case insensitivity for free. This makes the code much simpler.

For future reference, if there's an easy way to implement assignment operations, I'm pretty sure I could add support for non-environmental variables (the semantics are subtly different). And after that, implementing `export` would be pretty easy.